singularity-eos/eos/eos_gruneisen.cpp

```diff
@@ -12,12 +12,155 @@
 // publicly and display publicly, and to permit others to do so.
 //------------------------------------------------------------------------------
 
+#include <sstream>
+
+#include <root-finding-1d/root_finding.hpp>
 #include <singularity-eos/eos/eos.hpp>
 
 namespace singularity {
 
 PORTABLE_INLINE_FUNCTION Real square(const Real x) { return x * x; }
 PORTABLE_INLINE_FUNCTION Real cube(const Real x) { return x * x * x; }
+PORTABLE_INLINE_FUNCTION Real find_min_bounded_val(const Real root1, const Real root2,
+                                                   const Real min, const Real max) {
+  // Try to find the minimum bounded value. If none exists, return a value less than the
+  // minimum
+  const Real minroot = std::min(root1, root2);
+  const Real maxroot = std::max(root1, root2);
+  if (minroot > min && minroot < max) {
+    return minroot;
+  } else if (maxroot > min && maxroot < max) {
+    return maxroot;
+  } else {
+    return min - 1.;
+  }
+}
+/*
+The Gruneisen EOS diverges at a specific compression. Ensure that the maximum density is
+below the smallest singularity in the reference pressure curve
+*/
+PORTABLE_FUNCTION Real Gruneisen::GetRhoMax(const Real s1, const Real s2, const Real s3,
```

Comment:
It occurs to me that querying what rhomax is from the eos might be desirable but we don't want to re-do thhis calculation. I suggest renaming this `ComputeRhoMax` but exposing a method that return the value stored in the class for debugging purposes.